gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageTest.java

```diff
@@ -262,6 +274,95 @@ public void testCreateObjectWithGenerationId() throws Exception {
     }
   }
 
+  /** Test the Statistics of http GET request */
+  @Test
+  public void testHttpRequestStat() throws IOException {
+
+    MockHttpTransport transport =
+        mockTransport(
+            jsonDataResponse(newStorageObject(BUCKET_NAME, OBJECT_NAME)),
+            jsonErrorResponse(ErrorResponses.NOT_FOUND));
+
+    GoogleCloudStorage gcs = mockedGcs(GCS_OPTIONS, transport, trackingRequestInitializer);
+    WritableByteChannel writeChannel = gcs.create(RESOURCE_ID);
+    assertThat(writeChannel.isOpen()).isTrue();
+    assertThat(trackingRequestInitializer).isNotNull();
+    assertThat(httpStatistics.get(GoogleCloudStorageStatistics.ACTION_HTTP_GET_REQUEST).longValue())
+        .isEqualTo(1L);
+  }
+
+  /** Test the Failure Statistics of http GET & POST request */
+  @Test
+  public void testHttpGetRequestFailureStat() throws Exception {
```

Comment:
Refer other tests for assertions. 
Ex : `reupload_success_multipleWrites_singleUploadChunk` has assertions for exact http calls made, we could copy the same method and add assertions on http calls via stats

Reply:
Updated